unittests/allplatformstests.py

```diff
@@ -5011,7 +5011,7 @@ def test_c_cpp_stds(self):
             self.setconf('-Dc_std=c++11')
         env = get_fake_env()
         cc = detect_c_compiler(env, MachineChoice.HOST)
-        if cc.get_id() == 'msvc':
+        if cc.get_id() in {'msvc', 'clang-cl'}:
```

Comment:
```
unit tests: make the test_c_cpp_stds test more robust

Check clang-cl as well as cl, and clang as well as gcc. Also raise a
skip rather than a pass if the compiler is not supported
```

Where's the skip?

Reply:
Oh, I ended up dropping that because it's a weird test that does some testing that it expects to always work, and then some more testing that only works if there's a valid compiler. I'll update the commit message

Reply:
Ah ok, thanks for clarifying.

Reply:
It would be nice we had a reasonable test framework that had subtests that can skip :/